Client/Frontend/Browser/QRCodeViewController.swift

```diff
@@ -252,6 +254,30 @@ class QRCodeViewController: UIViewController, AVCaptureMetadataOutputObjectsDele
     }
 }
 
+extension QRCodeViewController: AVCaptureMetadataOutputObjectsDelegate {
+    func captureOutput(_ captureOutput: AVCaptureOutput!, didOutputMetadataObjects metadataObjects: [Any]!, from connection: AVCaptureConnection!) {
+        if metadataObjects == nil || metadataObjects.count == 0 {
+            self.captureSession.stopRunning()
+            let alert = UIAlertController(title: "", message: NSLocalizedString("The data is invalid", comment: "Text of the prompt user the data is invalid"), preferredStyle: .alert)
+            alert.addAction(UIAlertAction(title: NSLocalizedString("OK", comment: "OK button"), style: .default, handler: { (UIAlertAction) in
+                self.captureSession.startRunning()
+            }))
+            self.present(alert, animated: true, completion: nil)
+        } else {
+            self.captureSession.stopRunning()
+            stopScanLineAnimation()
+            self.dismiss(animated: true, completion: {
+                guard let metaData = metadataObjects.first as? AVMetadataMachineReadableCodeObject,
```

Comment:
I find in a case like this, multi lines makes it kind of hard to follow. It looks like `let qrCodeDelegate = self.qrCodeDelegate` is inside the guard's else statement to me.